Add a variant of ridgeplot example that uses sub-coordinates

mattpap opened the pull request:
Keeps the original look, but allows to preserve actual data (see `data (x, y)` in the tooltip).

![image](https://user-images.githubusercontent.com/27475/161083399-e4ab90fb-8e34-4923-adce-e0472fdce6ff.png)

tcmetzger commented:
That's a good point! I agree that we shouldn't use something that looks the same to illustrate two different concepts in two different places. I see basically the following options:

* we create a new and different example to illustrate categorical offsets in https://docs.bokeh.org/en/latest/docs/user_guide/categorical.html#categorical-offsets
* we find a way to make examples/plotting/file/ridgeplot.py use categorical offsets and thus be a valid example again for the 'categorical offsets' section in the user guide.

It sounds to me like using sub-coordinates is now the definitive way to go for ridge plots, so that means we should probably look at option one: creating a new example to serve as an illustration of categorical offsets. Do you have any ideas what that could be? If not, we can also just leave it at one example (the current fruit count example) in the user guide's 'categorical offsets' section and just not have a "more sophisticated example".

mattpap commented:
I took a different approach with this. I added a new category to `examples/topics`: `subcoordinates`, where I added a variant of `ridgeplot.py` that uses sub-coordinates. Whereas I reverted changes to `examples/topics/categorical`, while updating descriptions. This allows ridgeplot to be used for showing and comparing both approaches. I will continue in PR #13346, where I will add more examples and add documentation for sub-coordinates and update documentation for categoricals (pointing to sub-coordinates and explaining differences between both).

bryevdv commented:
@mattpap There is intended to be a 1-1 correspondence between `examples/topics/foo` directories, and subchapters under the Topics chapter of the user's guide. Do you actually envision an entire subchapter for this? If there aren't at least 3-4 subheadings and examples to flesh out a sub-chapter then it would be better to find an existing diretory corresponding to and existing chapter to add it to. Or of you don't want it to be in the User's guide, then it can go in one of the "grab bag" examples dirs that don't correspond to the docs hierarchy e.g. `examples/models` or `examples/plotting`


Edit: Here is the current chapter/subchapter hierarchy with the top level directories noted:

[Basic plotting](https://docs.bokeh.org/en/latest/docs/user_guide/basic.html)  `# examples/basic`

- [Scatter plots](https://docs.bokeh.org/en/latest/docs/user_guide/basic/scatters.html)
- [Lines and curves](https://docs.bokeh.org/en/latest/docs/user_guide/basic/lines.html)
- [Data sources](https://docs.bokeh.org/en/latest/docs/user_guide/basic/data.html)
- [Ranges and axes](https://docs.bokeh.org/en/latest/docs/user_guide/basic/axes.html)
- [Bar charts](https://docs.bokeh.org/en/latest/docs/user_guide/basic/bars.html)
- [Area glyphs](https://docs.bokeh.org/en/latest/docs/user_guide/basic/areas.html)
- [Grids and layouts](https://docs.bokeh.org/en/latest/docs/user_guide/basic/layouts.html)
- [Annotations](https://docs.bokeh.org/en/latest/docs/user_guide/basic/annotations.html)

[Appearance](https://docs.bokeh.org/en/latest/docs/user_guide/styling.html) `# examples/styling`

- [General visual properties](https://docs.bokeh.org/en/latest/docs/user_guide/styling/visuals.html)
- [Styling plot elements](https://docs.bokeh.org/en/latest/docs/user_guide/styling/plots.html)
- [Mathematical notation](https://docs.bokeh.org/en/latest/docs/user_guide/styling/mathtext.html)
- [Themes](https://docs.bokeh.org/en/latest/docs/user_guide/styling/themes.html)
- [Styling DOM elements](https://docs.bokeh.org/en/latest/docs/user_guide/styling/dom.html)

[Topic guides](https://docs.bokeh.org/en/latest/docs/user_guide/topics.html) `# examples/topics`

- [Images and raster data](https://docs.bokeh.org/en/latest/docs/user_guide/topics/images.html)
- [Contour plots](https://docs.bokeh.org/en/latest/docs/user_guide/topics/contour.html)
- [Hex tiles](https://docs.bokeh.org/en/latest/docs/user_guide/topics/hex.html)
- [Categorical plots](https://docs.bokeh.org/en/latest/docs/user_guide/topics/categorical.html)
- [Hierarchical data](https://docs.bokeh.org/en/latest/docs/user_guide/topics/hierarchical.html)
- [Geographical data](https://docs.bokeh.org/en/latest/docs/user_guide/topics/geo.html)
- [Network graphs](https://docs.bokeh.org/en/latest/docs/user_guide/topics/graph.html)
- [Timeseries plots](https://docs.bokeh.org/en/latest/docs/user_guide/topics/timeseries.html)
- [Pie and donut charts](https://docs.bokeh.org/en/latest/docs/user_guide/topics/pie.html)
- [Statistical plots](https://docs.bokeh.org/en/latest/docs/user_guide/topics/stats.html)

I could imaging *Basic Plotting -> Sub-coordinates* as making sense, or it could potentially go under *Appearance* if we wanted to broaden the scope of that chapter (cc @tcmetzger) but if you want to punt on this for now just move it to one of the grab-bag dirs and open an issue to discuss future docs work around it.

bryevdv commented:
@mattpap for expediency I think the simplest thing to do is to move the example to `examples/plotting` for now, to get this merged, and open an issue for future docs work. As you mention this really merits a full discussion/comparison so we'll just need to figure out the best place to land that.

bryevdv commented on examples/topics/categorical/ridgeplot.py:
```diff
@@ -1,13 +1,21 @@
-to the prompt *What probability would you assign to the phrase "Highly likely"*.
+
+This example demonstrates the uses of categorical offsets to position categorical
+values explicitly, which in this case allows for "poor-man's" sub-plots. This is
```
Avoiding idioms probably better for more people
```suggestion
values explicitly, which in this case allows for makeshift sub-plots. This is
```

bryevdv commented on examples/topics/categorical/ridgeplot.py:
```diff
@@ -1,13 +1,21 @@
+because the real data is not available for presentation, so for example it's not
+possible to show ridge line values in hover tool.
+
+Thus a better alternative is to use sub-coordinates as demonstrated in example
```
```suggestion
A better alternative is to use sub-coordinates as demonstrated in example
```

bryevdv reviewed (approved):
Just a couple of wording suggestions
